File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.js
const input = require("fs").readFileSync(0).toString().trim();
const N = Number(input);

let recursionAnswer = 0;
const recursion = (n) => {
  if (n === 1 || n === 2) {
    recursionAnswer++;
    return 1;
  }
  return recursion(n - 1) + recursion(n - 2);
};

let dpAnswer = 0;
const dp = (n) => {
  const fib = [1, 1];
  for (let i = 2; i < N; i++) {
    fib.push(fib[i - 1] + fib[i - 2]);
    dpAnswer++;
  }
  return fib[n - 1];
};

recursion(N);
dp(N);

console.log(`${recursionAnswer} ${dpAnswer}`);

File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.py
import sys
input = sys.stdin.readline

N = int(input())

fibo = [0] * (N + 1)  
fibo[1] = 1
fibo[2] = 1
for i in range(3, N + 1):
    fibo[i] = fibo[i - 1] + fibo[i - 2]

recursion_answer = fibo[N]  # if n == 1 or n == 2 실행 횟수
dp_answer = N - 2

print(f"{recursion_answer} {dp_answer}")